src/crimson/mgr/client.cc

```diff
@@ -62,6 +62,7 @@ seastar::future<> Client::ms_handle_reset(crimson::net::ConnectionRef c)
 {
   if (conn == c) {
     conn = nullptr;
+    tick_timer.cancel();
```

Comment:
@tchaikov Hi, why is the tick_timer canceled here? It seems that if we cancel the tick_timer here and the connection with mgr is down, there would be no way the osd can get connected to mgr again. I just submitted a pr for this: [33833](https://github.com/ceph/ceph/pull/33833), please take a look, thanks:-)

Reply:
1. a mgr client sends MMgrOpen to mgr when it gets connected.
1. mgr sends an MMgrConfigure to mgr client when handling MMgrOpen from that client or when the mgr configuration changes
1. there is no need to send updates to mgr when the connection to mgr is not established yet.

Reply:
Chances are that, when mgr response with MMgrConfigure, it may not be ready for a stable connection with the osd as it has not receive the osd's metadata from Monitor yet, which would lead mgr to reset the connection with the osd. The mgr expects the osd to reconnect later.

Reply:
> 3. there is no need to send updates to mgr when the connection to mgr is not established yet.

Seems to me the problem is that the connection policy to Mgr is lossy, which means it won't heal by itself. And the `reconnect()` (called by `tick()`) is also cancelled when the connection is lost, nothing will establish the connection again unless the configuration changes.

Reply:
> Chances are that, when mgr response with MMgrConfigure, it may not be ready for a stable connection with the osd as it has not received the osd's metadata from Monitor yet, 

i am not sure if i am missing anything. but by inspecting https://github.com/ceph/ceph/blob/eb488f8f6731d4bd03bab07853db4dec990350a1/src/mgr/DaemonServer.cc#L403  and https://github.com/ceph/ceph/blob/eb488f8f6731d4bd03bab07853db4dec990350a1/src/mgr/DaemonServer.cc#L2859 , i cannot find how metadata comes into play when mgr sends MMgrConfigure.

Reply:
This is where the metadata comes into play,  right after _send_configure():
```
  _send_configure(con);

  DaemonStatePtr daemon;
  if (daemon_state.exists(key)) {
    dout(20) << "updating existing DaemonState for " << key << dendl;
    daemon = daemon_state.get(key);
  }
  if (!daemon) {
    if (m->service_daemon) {
      dout(4) << "constructing new DaemonState for " << key << dendl;
      daemon = std::make_shared<DaemonState>(daemon_state.types);
      daemon->key = key;
      daemon->service_daemon = true;
      daemon_state.insert(daemon);
    } else {
      /* A normal Ceph daemon has connected but we are or should be waiting on
       * metadata for it. Close the session so that it tries to reconnect.
       */
      dout(2) << "ignoring open from " << key << " " << con->get_peer_addr()
              << "; not ready for session (expect reconnect)" << dendl;
      con->mark_down();
      return true;
    }
  }
```
The daemon_state is actually a set of daemon metadata including the osd metadata